### PR TITLE
ASK-0039: Add DK004 lint rule for bare except Exception: pass

### PR DESCRIPTION
**Pull request by movito**
## Summary
- Add new pattern lint rule **DK004** to `scripts/pattern_lint.py` that detects `except Exception: pass` and `except BaseException: pass` patterns that silently swallow errors
- 11 TDD tests covering positive detection, negative cases (logged, re-raised, returned, specific exceptions), noqa suppression, and edge cases (bare except, tuple exceptions)
- Fixed a real DK004 violation found in `scripts/validate_task_status.py` — narrowed `except Exception` to `except (OSError, UnicodeDecodeError)`
- **Workflow verification task**: exercises all 9 phases of the v3 feature-developer workflow

## Changes
| File | Change |
|------|--------|
| `scripts/pattern_lint.py` | `check_dk004()` + `_is_swallowed()` helper, registered in `lint_file()`, updated docstring |
| `tests/test_pattern_lint.py` | `TestDK004` class (11 tests) + updated integration tests |
| `scripts/validate_task_status.py` | Narrowed broad `except Exception` to specific exception types |

## Test Plan
- [x] 31 pattern_lint tests pass (11 new DK004 + 20 existing)
- [x] Full suite: 174 passed, 1 skipped
- [x] Pattern lint clean on existing codebase
- [x] Local CI (`ci-check.sh`) passes: Black, isort, flake8, pattern lint, pytest with 84% coverage

Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Adds a new AST-based lint rule that can fail CI/pre-commit across the repo, so the main risk is introducing false positives/negatives that block developer workflows or miss real swallowed exceptions.
> 
> **Overview**
> Adds a new pattern-lint rule **`DK004`** to `scripts/pattern_lint.py` that flags `except Exception`/`except BaseException` handlers whose bodies are empty or `pass`, with support for per-line suppression via `# noqa: DK004` and registration in `lint_file()`.
> 
> Expands `tests/test_pattern_lint.py` with a dedicated DK004 test suite and integration coverage for common allowed/blocked patterns (logged, re-raised, explicit return, specific exceptions, bare `except`, tuple-typed handlers).
> 
> Fixes an existing violation by narrowing `scripts/validate_task_status.py` from `except Exception:` to `except (OSError, UnicodeDecodeError):` when reading task files.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0bc764fe6b291ace110dfe44427dc19de3a153df. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: scripts/pattern_lint.py [hunk 1]
```diff
@@ -190,6 +191,94 @@ def check_dk003(tree: ast.AST, source_lines: list[str], path: str) -> list[Viola
...
+        if isinstance(stmt, ast.Return):
+            return False
+        if isinstance(stmt, ast.Expr) and isinstance(stmt.value, ast.Call):
+            func = stmt.value.func
+            # logger.error(...), logging.warning(...), etc.
+            if isinstance(func, ast.Attribute) and isinstance(func.value, ast.Name):
+                if func.value.id in ("logger", "logging", "log"):
+                    return False
+
+    return False
```
[the remark this answers is not in the document]
Comment (movito) [reply scripts/pattern_lint.py]: Fixed in a8c1d69 — simplified `_is_swallowed()` to remove the dead code loop. Now uses `return all(isinstance(stmt, ast.Pass) for stmt in body)` which is both correct and concise.

File: .adversarial/inputs/ASK-0039-code-review-input.md [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+## Valid Values and Boundaries
+
+- **`broad_exceptions` set**: `{"Exception", "BaseException"}` — exact string match against `ast.Name.id`
+- **`node.type`**: can be `None` (bare except), `ast.Name` (single type), `ast.Tuple` (multiple types), `ast.Attribute` (module.Type)
+- **`_is_swallowed` body**: can be empty list, list of `ast.Pass`, or list of meaningful statements
+
+## Key Questions
+
+1. Does `_is_swallowed` correctly handle an except body that has BOTH a logging call AND a pass statement (e.g., `logger.error(e); pass`)?
+2. Should the rule detect `except Exception` where the body is just a comment (AST strips comments, so `except Exception: # intentional\n    pass` still has `[ast.Pass]` as body)?
```
[the remark this answers is not in the document]
Comment (movito) [reply .adversarial/inputs/ASK-0039-code-review-input.md]: Moot after a8c1d69 — the simplified `_is_swallowed()` no longer has a loop or logger recognition. It just checks `all(isinstance(stmt, ast.Pass) for stmt in body)`. Any non-pass statement (logging, assignments, function calls) correctly indicates the exception is handled.

File: .agent-context/reviews/ASK-0039-evaluator-review.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+   (e.g., some_logger.warn(e)).
+ • What happens: The function sees these calls as no-ops, concludes the body is
+   effectively swallowing the exception, and flags them. This is a false
+   positive scenario if the developer used a custom logger name.
+ • Expected: Possibly a more thorough check or a note clarifying custom loggers
+   are not recognized.
+ • Test coverage: NOT covered. The tests only show standard “logger” or
+   “logging.”
+ • Severity: Latent (acceptable if using a known logger, but breaks if a custom
+   logger is used).
```
[the remark this answers is not in the document]
Comment (movito) [reply .agent-context/reviews/ASK-0039-evaluator-review.md]: Fixed in a8c1d69. The simplified `_is_swallowed()` no longer hardcodes logger names. It now uses `all(isinstance(stmt, ast.Pass) for stmt in body)` — any non-`pass` statement (including custom logger calls like `app_logger.warning(e)`, `mylogger.error(...)`) correctly indicates the exception is being handled. The `# noqa: DK004` escape hatch also remains available.

File: .agent-context/reviews/ASK-0039-evaluator-review.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+ with logging + pass                                  “handled”
+
+
+Verdict
+
+CONCERNS. While no immediate breakage is identified, there are latent
+correctness issues and test coverage gaps in the new DK004 logic. Specifically,
+ignoring tuple-based broad exceptions and requiring a recognized logger name
+pose risks. Additional tests or minor logic enhancements could address these
+issues before they cause real-world escapes.
```
[the remark this answers is not in the document]
Comment (movito) [reply .agent-context/reviews/ASK-0039-evaluator-review.md]: Won't fix — this is a comment on the evaluator's output format, not a code issue. The evaluator review is an informational artifact that was consumed by the feature developer to guide implementation decisions.
